app/assets/javascripts/analytics-ga4/ga4-finder-tracker.js

```diff
@@ -0,0 +1,140 @@
+//= require govuk/vendor/polyfills/Element/prototype/closest.js
+;(function (global) {
+  'use strict'
+
+  var GOVUK = global.GOVUK || {}
+  GOVUK.analyticsGa4 = GOVUK.analyticsGa4 || {}
+
+  GOVUK.analyticsGa4.Ga4FinderTracker = {
+
+    setFilterIndexes: function () {
+      var filterContainer = document.querySelector('[data-ga4-filter-container]')
+
+      if (!filterContainer) {
+        return
+      }
+
+      var filterSections = filterContainer.querySelectorAll('[data-ga4-section]')
+
+      for (var i = 0; i < filterSections.length; i++) {
+        filterSections[i].setAttribute('data-ga4-index', JSON.stringify({ index_section: i + 1, index_section_count: filterSections.length }))
+      }
+    },
+
+    // changeEventMetadata is a string referring to the type of form change and the element type that triggered it. For example 'update-filter checkbox'.
+    trackChangeEvent: function (eventTarget, changeEventMetadata) {
+      var PIIRemover = new window.GOVUK.analyticsGa4.PIIRemover()
+
+      changeEventMetadata = changeEventMetadata.split(' ')
+
+      var elementType = changeEventMetadata[1]
+
+      var section = eventTarget.closest('[data-ga4-section]')
+
+      var schema = new GOVUK.analyticsGa4.Schemas().eventSchema()
+      schema.event = 'event_data'
+      schema.event_data.type = 'finder'
+
+      var elementInfo = this.getElementInfo(eventTarget, elementType, section)
+      var elementValue = elementInfo.elementValue
+      var wasFilterRemoved = elementInfo.wasFilterRemoved
+
+      schema.event_data.text = elementValue
+
+      var changeType = changeEventMetadata[0]
+      switch (changeType) {
+        case 'update-filter':
+          schema.event_data.event_name = 'select_content'
+
+          if (section) {
+            schema.event_data.section = section.getAttribute('data-ga4-section')
+          }
+
+          if (wasFilterRemoved) {
+            schema.event_data.action = 'remove'
+            schema.event_data.text = elementType === 'text' ? undefined : elementValue
+          } else {
+            schema.event_data.action = elementType === 'text' ? 'search' : 'select'
+            schema.event_data.index = this.getSectionIndex(section)
+          }
+          break
+
+        case 'update-keyword':
+          schema.event_data.event_name = 'search'
```

Comment:
It looks like all of these `case` statements set `event_name` to `select_content`, apart from this one. Maybe set it to `select_content` outside of the switch statement, then just change it in this case?

Reply:
Done as part of moving this code to a separate function :+1: